### ISSUE TITLE
Null reference when calling Trim() on interpolated string containing null property

### ISSUE THREAD

**Issue by G1Vh** [nhibernate/nhibernate-core#2860]
Linq query worked in 5.3.8 but fails in 5.3.9 with null reference exception.

We have code like this:
```
var employees = await employeeRepository.Query()
                                                    .Select(c => new Employee
                                                     {
                                                         Id = c.Id,
                                                         Name = $"{c.Name} {c.Surname}".Trim()
                                                     })
                                                    .ToListAsync();
```

One of employees has null in Surname property.
If `Trim()` removed query works in 5.3.9.

Generated sql is same in both versions:
```
    select
        employee0_.id as col_0_0_,
        ?p0 as col_1_0_,
        employee0_.name as col_2_0_,
        employee0_.surname as col_3_0_
    from
        employee employee0_;
    ?p0 = '{0} {1}' [Type: String (7:0:0)]
```

**Comment by bahusoid** [comment]
Can you share full exception stacktrace?

**Comment by G1Vh** [comment]
```
NHibernate.Exceptions.GenericADOException: could not execute query
[ select employee0_.id as col_0_0_, ?p0 as col_1_0_, employee0_.name as col_2_0_, employee0_.surname as col_3_0_ from employee employee0_ ]
  Name:p1 - Value:{0} {1}
[SQL: select employee0_.id as col_0_0_, ?p0 as col_1_0_, employee0_.name as col_2_0_, employee0_.surname as col_3_0_ from employee employee0_]
 ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at NHibernate.Loader.Hql.QueryLoader.GetResultRowAsync(Object[] row, DbDataReader rs, ISessionImplementor session, CancellationToken cancellationToken)
   at NHibernate.Loader.Hql.QueryLoader.GetResultColumnOrRowAsync(Object[] row, IResultTransformer resultTransformer, DbDataReader rs, ISessionImplementor session, CancellationToken cancellationToken)
   at NHibernate.Loader.Loader.GetRowFromResultSetAsync(DbDataReader resultSet, ISessionImplementor session, QueryParameters queryParameters, LockMode[] lockModeArray, EntityKey optionalObjectKey, IList hydratedObjects, EntityKey[] keys, Boolean returnProxies, IResultTransformer forcedResultTransformer, QueryCacheResultBuilder queryCacheResultBuilder, Action`2 cacheBatchingHandler, CancellationToken cancellationToken)
   at NHibernate.Loader.Loader.DoQueryAsync(ISessionImplementor session, QueryParameters queryParameters, Boolean returnProxies, IResultTransformer forcedResultTransformer, QueryCacheResultBuilder queryCacheResultBuilder, CancellationToken cancellationToken)
   at NHibernate.Loader.Loader.DoQueryAndInitializeNonLazyCollectionsAsync(ISessionImplementor session, QueryParameters queryParameters, Boolean returnProxies, IResultTransformer forcedResultTransformer, QueryCacheResultBuilder queryCacheResultBuilder, CancellationToken cancellationToken)
   at NHibernate.Loader.Loader.DoListAsync(ISessionImplementor session, QueryParameters queryParameters, IResultTransformer forcedResultTransformer, QueryCacheResultBuilder queryCacheResultBuilder, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at NHibernate.Loader.Loader.DoListAsync(ISessionImplementor session, QueryParameters queryParameters, IResultTransformer forcedResultTransformer, QueryCacheResultBuilder queryCacheResultBuilder, CancellationToken cancellationToken)
   at NHibernate.Loader.Loader.ListIgnoreQueryCacheAsync(ISessionImplementor session, QueryParameters queryParameters, CancellationToken cancellationToken)
   at NHibernate.Hql.Ast.ANTLR.QueryTranslatorImpl.ListAsync(ISessionImplementor session, QueryParameters queryParameters, CancellationToken cancellationToken)
   at NHibernate.Engine.Query.HQLQueryPlan.PerformListAsync(QueryParameters queryParameters, ISessionImplementor session, IList results, CancellationToken cancellationToken)
   at NHibernate.Impl.SessionImpl.ListAsync(IQueryExpression queryExpression, QueryParameters queryParameters, IList results, Object filterConnection, CancellationToken cancellationToken)
   at NHibernate.Impl.SessionImpl.ListAsync(IQueryExpression queryExpression, QueryParameters queryParameters, IList results, Object filterConnection, CancellationToken cancellationToken)
   at NHibernate.Impl.AbstractSessionImpl.ListAsync[T](IQueryExpression query, QueryParameters parameters, CancellationToken cancellationToken)
   at NHibernate.Impl.AbstractQueryImpl2.ListAsync[T](CancellationToken cancellationToken)
   at NHibernate.Linq.DefaultQueryProvider.ExecuteListAsync[TResult](Expression expression, CancellationToken cancellationToken)
   at NHibernate.Linq.LinqExtensionMethods.<>c__DisplayClass59_0`1.<<ToListAsync>g__InternalToListAsync|0>d.MoveNext()
--- End of stack trace from previous location ---
   at CRM.Controllers.MainController.Index() in C:\Users\g1vh\Desktop\proj\CRM\Controllers\MainController.cs:line 79
```

**Comment by fredericDelaporte** [comment]
Fixed by #2875
